tests/pq/test_pgconn.py

```diff
@@ -389,6 +393,100 @@ def test_set_single_row_mode(pgconn):
     pgconn.set_single_row_mode()
 
 
+@contextlib.contextmanager
+def cancellable_query(pgconn: PGconn, monitor_conn: PGconn) -> Iterator[None]:
+    pgconn.send_query_params(b"SELECT pg_sleep($1)", [b"180"])
```

Comment:
Wouldn't be better to define this helper using `psycopg.Connection`? If you did it this way because you have found most of the tests in the tests/pq suite using only libpq objects, it's because when they were written psycopg wasn't implemented yet :smile: